feat: adds German UI translations

trueberryless opened the pull request:
add section in translations.ts for German translation

trueberryless commented:
Wait, how cool is this automation of the documentation?!?

![image](https://github.com/user-attachments/assets/92424259-e9c0-422e-8816-cb0ce5685675)

This guy is INSANELY smart again, love it!

Ergberg commented on packages/starlight-blog/translations.ts:
```diff
@@ -37,4 +37,23 @@ export const Translations = {
+  de: {
+    'starlightBlog.authors.count_one': '{{count}} Beitrag von {{author}}',
+    'starlightBlog.authors.count_other': '{{count}} Beiträge von {{author}}',
+    'starlightBlog.pagination.prev': 'Vorheriger Beitrag',
+    'starlightBlog.pagination.next': 'Nächster Beitrag',
```
```suggestion
    'starlightBlog.pagination.prev': 'Neuere Beiträge',
    'starlightBlog.pagination.next': 'Ältere Beiträge',
```

Ergberg reviewed (commented):
Hi @trueberryless,

das ist ja toll, danke für die Übersetzung!

even if they are called "prev" and "next", I guess "newer" and "older" better fits the meaning chosen in the other languages.

For "featured", I wondered if "empfohlener Beitrag" would sound more natural, but have no strong opinion (and in deed it is not "recommended").

trueberryless commented:
> For "featured", I wondered if "empfohlener Beitrag" would sound more natural, but have no strong opinion (and in deed it is not "recommended").  

Yeah, initially I thought this too, then I changed it because this exact reason, but now that you confirmed that it sounds more natural, I changed it back to "empfohlen"...

Thanks for the tip with prev and next, I confused them with the prev and next buttons on single blog post sites... my bad

HiDeoo reviewed (approved):
Amazing, thanks you both for the translations and the reviews :star2:
